src/nbl/builtin/template/CArchive.h.in

```diff
@@ -0,0 +1,34 @@
+#ifndef _@_GUARD_SUFFIX_@_C_ARCHIVE_H_
+#define _@_GUARD_SUFFIX_@_C_ARCHIVE_H_
+
+#include "nbl/system/CFileArchive.h"
+#include "@NBL_BS_HEADER_FILENAME@"
+
+namespace @_NAMESPACE_@
+{
+	constexpr std::string_view pathPrefix = "@_BUNDLE_ARCHIVE_ABSOLUTE_PATH_@";
+	constexpr bool hasPathPrefix(std::string_view str) { return str.find(pathPrefix) == 0ull; }
+
+	class CArchive final : public nbl::system::CFileArchive
+	{
+		public:
+			inline CArchive(nbl::system::logger_opt_smart_ptr&& logger)
+				: nbl::system::CFileArchive(nbl::system::path(pathPrefix.data()),std::move(logger), resources) {}
+			
+		protected:
+			file_buffer_t getFileBuffer(const nbl::system::IFileArchive::SListEntry* item) override
+			{
+					auto found = get_resource_runtime(item->pathRelativeToArchive.string());
+					return { const_cast<uint8_t*>(found.first), found.second, nullptr };
+			}
+
+			inline static const nbl::core::vector<nbl::system::IFileArchive::SListEntry> resources
+			{
+@_RESOURCES_INIT_LIST_@
+			};			
```

Comment:
this static better move to the builtin resource `.cpp`, so you don't get duplicate symbols in everything that includes it (also so that its a static global outside of a class, such that the symbol is hidden, and that it can be created on the heap with `new` which is what the later `shared_ptr` will need)

This also means that the constructor of `CArchvie` needs to be `BR_API` and defined in the `.cpp` as well

Reply:
sure, so let's use one library for builtin & archive data

Reply:
resolved